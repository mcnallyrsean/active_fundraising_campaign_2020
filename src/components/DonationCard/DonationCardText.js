import React from "react";
import PropTypes from "prop-types";
import converter from "number-to-words";

export default function DonationCardText({
  totalDonors,
  goalReached,
  daysRemaining,
}) {
  if (daysRemaining >= 0) {
    return (
      <div>
        {goalReached && (
          <h1>
            Goal reached! You still have{" "}
            {daysRemaining < 21
              ? `${converter.toWords(daysRemaining)} `
              : `${daysRemaining} `}
            day
            {daysRemaining > 1 ? "s" : ""} left to fund this project.
          </h1>
        )}
        {!goalReached && (
          <h1 style={{ margin: "0" }}>
            Only{" "}
            {daysRemaining < 21
              ? `${converter.toWords(daysRemaining)} `
              : `${daysRemaining} `}
            day
            {daysRemaining > 1 ? "s" : ""} left to fund this project.
          </h1>
        )}
        {totalDonors > 0 ? (
          <p className="donation-card-text">
            Join the <strong>{totalDonors}</strong> total donor
            {totalDonors > 1 ? "s" : ""} who
            {totalDonors > 1 ? " have" : " has"} already supported this project.
          </p>
        ) : (
          <p className="donation-card-text">
            Please consider supporting this project.
          </p>
        )}
      </div>
    );
  } else {
    return (
      <div>
        <h1 style={{ margin: "0" }}>The campaign is over.</h1>
        <p className="donation-card-text">Thank you to all who donated!</p>
      </div>
    );
  }
}

DonationCardText.propTypes = {
  totalDonors: PropTypes.number,
  goalReached: PropTypes.bool,
  daysRemaining: PropTypes.number,
};
