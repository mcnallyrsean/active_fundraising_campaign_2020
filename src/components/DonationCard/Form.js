import React, { useState } from "react";
import NumberFormat from "react-number-format";
import { useAlert } from "react-alert";

export default function Form({
  handleDonation,
  totalDonors,
  setTotalDonors,
  minimumDonation,
  daysRemaining,
}) {
  const alert = useAlert();
  const [inputAmount, setInputAmount] = useState("");
  const [formattedAmount, setFormattedAmount] = useState("");

  const handleChange = (value) => {
    setFormattedAmount(value.formattedValue);
    setInputAmount(value.floatValue);
  };

  const handleSubmit = (value) => {
    if (value >= minimumDonation) {
      handleDonation(value);
      setTotalDonors(totalDonors + 1);
      setInputAmount("");
      setFormattedAmount("");
      alert.success("Donation submitted. Thank you!");
    } else {
      alert.error(`You must donate at least $${minimumDonation}`);
    }
  };

  return (
    <div className="form">
      <label>
        <span className="aria-text">Donation Form:</span>
        <NumberFormat
          value={formattedAmount}
          displayType={"text"}
          thousandSeparator={true}
          decimalScale={2}
          prefix={"$"}
          placeholder={"$"}
          displayType="input"
          onValueChange={(value) => handleChange(value)}
          disabled={daysRemaining < 0 ? true : false}
        />
      </label>

      <button
        disabled={daysRemaining < 0 ? true : false}
        type="submit"
        onClick={() => handleSubmit(inputAmount)}
      >
        Give Now
      </button>
    </div>
  );
}
