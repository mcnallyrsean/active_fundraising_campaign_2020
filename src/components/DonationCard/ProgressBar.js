import React, { useEffect, useState } from "react";

export default function ProgressBar({ donationsTotal, goal }) {
  const [progress, setProgress] = useState(0);
  useEffect(() => {
    setProgress((donationsTotal / goal) * 100);
  }, [donationsTotal]);

  return (
    <div
      role="progressbar"
      aria-valuenow={progress}
      aria-valuemin="0"
      aria-valuemax="100"
      className="progress-bar"
    >
      <div className="progress" style={{ width: `${progress}%` }}></div>
      <span className="aria-text">{progress}%</span>
    </div>
  );
}
