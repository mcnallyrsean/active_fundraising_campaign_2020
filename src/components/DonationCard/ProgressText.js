import React from "react";
import NumberFormat from "react-number-format";
import PropTypes from "prop-types";

export default function ProgressText({ goal, amountToGoal, goalReached }) {
  return (
    <div className="progress-text-arrow">
      <p className="progress-text">
        <strong>
          <sup>$</sup>
          <NumberFormat
            value={Math.abs(amountToGoal)}
            displayType={"text"}
            thousandSeparator={true}
            decimalScale={2}
          />
        </strong>
        {goalReached
          ? " over the project goal!"
          : " still needed to fund the project."}
      </p>
    </div>
  );
}

ProgressText.propTypes = {
  goal: PropTypes.number,
  goalReached: PropTypes.bool,
  amountToGoal: PropTypes.number,
};
