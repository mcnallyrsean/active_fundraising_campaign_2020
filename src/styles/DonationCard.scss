@import "./colors.scss";

$border-radius: 10px;

.card-container {
  width: 32.5rem;
  @media all and (max-width: 990px) {
    width: 90vw;
  }
}

.card-top {
  background-color: $ocean-30;
  border-radius: $border-radius;
  margin-bottom: 15px;
}

.progress-text-arrow {
  position: relative;
  z-index: 1;
  padding: 10px;
  &:before {
    bottom: -10px;
    right: 20px;
    content: "";
    position: absolute;
    width: 0;
    height: 0;
    z-index: 0;
    border-left: 15px solid transparent;
    border-right: 15px solid transparent;
    border-top: 15px solid $ocean-30;
  }
}

.progress-text {
  color: $ocean-80;
  margin: 0;
}

.card-main {
  background: $white;
  border-radius: $border-radius;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 60px 40px;
  position: relative;
  min-height: 15rem;
  box-shadow: 0px 7.5px 7.5px $slate-30;
  @media all and (max-width: 990px) {
    padding: 30px;
  }
}

.progress-bar {
  border-radius: $border-radius $border-radius 0 0;
  position: absolute;
  top: 0;
  left: 0;
  height: 15px;
  width: 100%;
  background: $mint-40;
  overflow: hidden;
}

.progress {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  background-color: $mint;
  transition: width 0.25s ease;
}

.donation-card-text {
  padding: 25px 0;
}

.form {
  display: flex;
  @media all and (max-width: 990px) {
    flex-direction: column;
  }
  label {
    flex: 1;
  }

  input {
    border-radius: $border-radius 0 0 $border-radius;
    padding: 15px 20px;
    width: 100%;
    border-top: 1px solid $slate-40;
    border-left: 1px solid $slate-40;
    border-bottom: 1px solid $slate-40;
    border-right: 0;
    font-size: 1.75rem;
    font-weight: 700;
    &:disabled {
      filter: opacity(0.5);
    }
    @media all and (max-width: 990px) {
      border: 1px solid $slate-40;
      border-radius: $border-radius;
    }
  }
  button {
    @media all and (max-width: 990px) {
      border-radius: $border-radius;
      padding: 15px;
      margin-top: 10px;
      line-height: 1.15;
    }
  }
}

.link-to-admin {
  position: fixed;
  right: 5%;
  bottom: 5%;
  @media all and (max-width: 990px) {
    left: 0;
    right: 0;
    margin: 0 auto;
  }
  @media all and (max-width: 450px) {
    position: relative;
    margin-top: 25px;
  }
}
.link-to-admin-cnt {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.aria-text {
  position: absolute;
  text-indent: -9999px;
}
