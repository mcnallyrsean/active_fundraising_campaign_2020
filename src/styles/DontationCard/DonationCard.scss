@import "../colors.scss";

$border-radius: 10px;

.card-container {
  width: 32.5rem;
}

.card-top {
  background-color: $ocean-30;
  border-radius: $border-radius;
  margin-bottom: 15px;
}

.progress-text-arrow {
  position: relative;
  z-index: 1;
  padding: 10px;
  &:before {
    bottom: -10px;
    right: 20px;
    content: "";
    position: absolute;
    width: 0;
    height: 0;
    z-index: 0;
    border-left: 15px solid transparent;
    border-right: 15px solid transparent;
    border-top: 15px solid $ocean-30;
  }
}

.progress-text {
  color: $ocean-80;
  margin: 0;
}

.card-main {
  background: $white;
  border-radius: $border-radius;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 60px 40px;
  position: relative;
  height: 15rem;
  box-shadow: 0px 7.5px 7.5px $slate-30;
}

.progress-bar {
  border-radius: $border-radius $border-radius 0 0;
  position: absolute;
  top: 0;
  left: 0;
  height: 15px;
  width: 100%;
  background: $mint-40;
  overflow: hidden;
}

.progress {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  background-color: $mint;
  transition: width 0.25s ease;
}

.donation-card-text {
  padding: 25px 0;
}

.form {
  display: flex;
  label {
    flex: 1;
  }
  input {
    border-radius: $border-radius 0 0 $border-radius;
    padding: 15px 20px;
    width: 100%;
    border-top: 1px solid $slate-40;
    border-left: 1px solid $slate-40;
    border-bottom: 1px solid $slate-40;
    border-right: 0;
    font-size: 1.75rem;
    font-weight: 700;
  }
  button {
    background-color: $mint;
    border: 0;
    border-radius: 0 $border-radius $border-radius 0;
    color: $white;
    font-family: "Poppins";
    font-weight: 700;
    font-size: 1.3rem;
    padding: 0px 20px;
  }
}

.aria-text {
  position: absolute;
  text-indent: -9999px;
}
