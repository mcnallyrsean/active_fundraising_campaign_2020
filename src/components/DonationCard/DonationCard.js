import React, { useState, useEffect } from "react";
import ProgressText from "./ProgressText";
import ProgressBar from "./ProgressBar";
import DonationCardText from "./DonationCardText";
import Form from "./Form";

export default function DonationCard() {
  const [goal, setGoal] = useState(5000.0);
  const [goalReached, setGoalReached] = useState(false);
  const [donationsTotal, setDonationsTotal] = useState(0);
  const [amountToGoal, setAmountToGoal] = useState(goal - donationsTotal);
  const [totalDonors, setTotalDonors] = useState(0);
  const [endDate, setEndDate] = useState(new Date(2020, 6, 2, 0, 0));

  const handleDonation = (amount) => {
    setDonationsTotal(donationsTotal + amount);
  };

  useEffect(() => {
    setAmountToGoal(goal - donationsTotal);
  }, [donationsTotal]);

  useEffect(() => {
    if (amountToGoal <= 0) {
      setGoalReached(true);
    }
  }, [amountToGoal]);

  return (
    <div className="card-container">
      <div className="card-top">
        <ProgressText
          goal={goal}
          amountToGoal={amountToGoal}
          goalReached={goalReached}
        ></ProgressText>
      </div>
      <div className="card-main">
        <ProgressBar donationsTotal={donationsTotal} goal={goal}></ProgressBar>
        <DonationCardText
          goalReached={goalReached}
          endDate={endDate}
          totalDonors={totalDonors}
        ></DonationCardText>
        <Form
          handleDonation={handleDonation}
          totalDonors={totalDonors}
          setTotalDonors={setTotalDonors}
        ></Form>
      </div>
    </div>
  );
}
