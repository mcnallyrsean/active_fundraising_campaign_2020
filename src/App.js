import React, { useEffect } from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";
import { AnimatedSwitch } from "react-router-transition";
import {
  addMonths,
  differenceInCalendarDays,
  isValid,
  parseISO,
} from "date-fns";
import { transitions, positions, Provider as AlertProvider } from "react-alert";
import AlertTemplate from "react-alert-template-basic";
import "./styles/app.scss";
import DonationCard from "./components/DonationCard/DonationCard";
import Admin from "./components/Admin/Admin";
import createPersistedState from "use-persisted-state";

const useGoalState = createPersistedState("goal");
const useMinimumDonationState = createPersistedState("minimumDonation");
const useEndDateState = createPersistedState("endDate");
const useDaysRemainingState = createPersistedState("daysRemaining");
const useGoalReachedState = createPersistedState("goalReached");
const useDonationsTotalState = createPersistedState("donationsTotal");
const useAmountToGoalState = createPersistedState("amountToGoal");
const useTotalDonorsState = createPersistedState("totalDonors");

export default function App() {
  const [goal, setGoal] = useGoalState(5000.0);
  const [minimumDonation, setMinimumDonation] = useMinimumDonationState(5.0);
  const [endDate, setEndDate] = useEndDateState(addMonths(new Date(), 1));
  const [daysRemaining, setDaysRemaining] = useDaysRemainingState(0);
  const [goalReached, setGoalReached] = useGoalReachedState(false);
  const [donationsTotal, setDonationsTotal] = useDonationsTotalState(0);
  const [amountToGoal, setAmountToGoal] = useAmountToGoalState(
    goal - donationsTotal
  );
  const [totalDonors, setTotalDonors] = useTotalDonorsState(0);

  useEffect(() => {
    setAmountToGoal(goal - donationsTotal);
  }, [goal, donationsTotal]);

  useEffect(() => {
    amountToGoal <= 0 ? setGoalReached(true) : setGoalReached(false);
  }, [goal, amountToGoal]);

  useEffect(() => {
    if (isValid(endDate)) {
      setDaysRemaining(differenceInCalendarDays(endDate, new Date()));
    } else {
      setDaysRemaining(differenceInCalendarDays(parseISO(endDate), new Date()));
    }
  }, [endDate]);

  const alertOptions = {
    position: positions.BOTTOM_CENTER,
    timeout: 5000,
    offset: "15px",
    transition: transitions.SCALE,
  };

  return (
    <AlertProvider template={AlertTemplate} {...alertOptions}>
      <Router>
        <AnimatedSwitch
          atEnter={{ opacity: 0 }}
          atLeave={{ opacity: 0 }}
          atActive={{ opacity: 1 }}
          className="switch-wrapper"
        >
          <Route exact path="/">
            <DonationCard
              goal={goal}
              endDate={endDate}
              goalReached={goalReached}
              donationsTotal={donationsTotal}
              amountToGoal={amountToGoal}
              totalDonors={totalDonors}
              setDonationsTotal={setDonationsTotal}
              setTotalDonors={setTotalDonors}
              minimumDonation={minimumDonation}
              daysRemaining={daysRemaining}
            />
          </Route>
          <Route path="/super-secret-admin-page">
            <Admin
              goal={goal}
              setGoal={setGoal}
              endDate={endDate}
              setEndDate={setEndDate}
              minimumDonation={minimumDonation}
              setMinimumDonation={setMinimumDonation}
              setTotalDonors={setTotalDonors}
              setDonationsTotal={setDonationsTotal}
              setAmountToGoal={setAmountToGoal}
            />
          </Route>
          <Route path="*">
            <DonationCard goal={goal} endDate={endDate} />
          </Route>
        </AnimatedSwitch>
      </Router>
    </AlertProvider>
  );
}
