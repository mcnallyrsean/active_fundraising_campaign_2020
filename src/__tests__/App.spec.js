import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import { Provider as AlertProvider } from "react-alert";
import AlertTemplate from "react-alert-template-basic";

import App from "../App";
import DonationCard from "../components/DonationCard/DonationCard";

describe("App", () => {
  test("snapshot renders", () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the inner DonationCard", () => {
    const wrapper = mount(<App />);
    expect(wrapper.find("DonationCard").length).toEqual(1);
  });
});

describe("DonationCard", () => {
  test("snapshot renders", () => {
    const component = renderer.create(
      <AlertProvider template={AlertTemplate}>
        <BrowserRouter>
          <DonationCard></DonationCard>
        </BrowserRouter>
      </AlertProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the inner Progress Text", () => {
    const wrapper = mount(
      <AlertProvider template={AlertTemplate}>
        <BrowserRouter>
          <DonationCard></DonationCard>
        </BrowserRouter>
      </AlertProvider>
    );
    expect(wrapper.find("ProgressText").length).toEqual(1);
  });

  it("renders the inner Progress Bar", () => {
    const wrapper = mount(
      <AlertProvider template={AlertTemplate}>
        <BrowserRouter>
          <DonationCard></DonationCard>
        </BrowserRouter>
      </AlertProvider>
    );
    expect(wrapper.find("ProgressBar").length).toEqual(1);
  });

  it("renders the inner Donation Card Text", () => {
    const wrapper = mount(
      <AlertProvider template={AlertTemplate}>
        <BrowserRouter>
          <DonationCard></DonationCard>
        </BrowserRouter>
      </AlertProvider>
    );
    expect(wrapper.find("DonationCardText").length).toEqual(1);
  });

  it("renders the inner Form", () => {
    const wrapper = mount(
      <AlertProvider template={AlertTemplate}>
        <BrowserRouter>
          <DonationCard></DonationCard>
        </BrowserRouter>
      </AlertProvider>
    );
    expect(wrapper.find("Form").length).toEqual(1);
  });
});
