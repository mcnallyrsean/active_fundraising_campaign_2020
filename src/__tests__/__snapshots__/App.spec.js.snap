// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="switch-wrapper"
>
  <div
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <div
      className="card-container"
    >
      <div
        className="card-top"
      >
        <div
          className="progress-text-arrow"
        >
          <p
            className="progress-text"
          >
            <strong>
              <sup>
                $
              </sup>
              <span>
                5,000
              </span>
            </strong>
             still needed to fund the project.
          </p>
        </div>
      </div>
      <div
        className="card-main"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow={0}
          className="progress-bar"
          role="progressbar"
        >
          <div
            className="progress"
            style={
              Object {
                "width": "0%",
              }
            }
          />
          <span
            className="aria-text"
          >
            0
            %
          </span>
        </div>
        <div>
          <h1
            style={
              Object {
                "margin": "0",
              }
            }
          >
            Only
             
            zero 
            day
            
             left to fund this project.
          </h1>
          <p
            className="donation-card-text"
          >
            Please consider supporting this project.
          </p>
        </div>
        <div
          className="form"
        >
          <label>
            <span
              className="aria-text"
            >
              Donation Form:
            </span>
            <input
              disabled={false}
              inputMode="numeric"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              placeholder="$"
              type="text"
              value=""
            />
          </label>
          <button
            disabled={false}
            onClick={[Function]}
            type="submit"
          >
            Give Now
          </button>
        </div>
        <div
          className="link-to-admin"
        >
          <a
            href="/super-secret-admin-page"
            onClick={[Function]}
          >
            <div
              className="link-to-admin-cnt"
            >
              <svg
                height="24"
                viewBox="0 0 492.004 492.004"
                width="24"
              >
                <path
                  d="M484.14 226.886L306.46 49.202c-5.072-5.072-11.832-7.856-19.04-7.856-7.216 0-13.972 2.788-19.044 7.856l-16.132 16.136c-5.068 5.064-7.86 11.828-7.86 19.04 0 7.208 2.792 14.2 7.86 19.264L355.9 207.526H26.58C11.732 207.526 0 219.15 0 234.002v22.812c0 14.852 11.732 27.648 26.58 27.648h330.496L252.248 388.926c-5.068 5.072-7.86 11.652-7.86 18.864 0 7.204 2.792 13.88 7.86 18.948l16.132 16.084c5.072 5.072 11.828 7.836 19.044 7.836 7.208 0 13.968-2.8 19.04-7.872l177.68-177.68c5.084-5.088 7.88-11.88 7.86-19.1.016-7.244-2.776-14.04-7.864-19.12z"
                />
              </svg>
              <span>
                Super Secret Admin Page
              </span>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DonationCard snapshot renders 1`] = `
<div
  className="card-container"
>
  <div
    className="card-top"
  >
    <div
      className="progress-text-arrow"
    >
      <p
        className="progress-text"
      >
        <strong>
          <sup>
            $
          </sup>
          <span>
            
          </span>
        </strong>
         still needed to fund the project.
      </p>
    </div>
  </div>
  <div
    className="card-main"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={NaN}
      className="progress-bar"
      role="progressbar"
    >
      <div
        className="progress"
        style={
          Object {
            "width": "NaN%",
          }
        }
      />
      <span
        className="aria-text"
      >
        NaN
        %
      </span>
    </div>
    <div>
      <h1
        style={
          Object {
            "margin": "0",
          }
        }
      >
        The campaign is over.
      </h1>
      <p
        className="donation-card-text"
      >
        Thank you to all who donated!
      </p>
    </div>
    <div
      className="form"
    >
      <label>
        <span
          className="aria-text"
        >
          Donation Form:
        </span>
        <input
          disabled={false}
          inputMode="numeric"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          placeholder="$"
          type="text"
          value=""
        />
      </label>
      <button
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Give Now
      </button>
    </div>
    <div
      className="link-to-admin"
    >
      <a
        href="/super-secret-admin-page"
        onClick={[Function]}
      >
        <div
          className="link-to-admin-cnt"
        >
          <svg
            height="24"
            viewBox="0 0 492.004 492.004"
            width="24"
          >
            <path
              d="M484.14 226.886L306.46 49.202c-5.072-5.072-11.832-7.856-19.04-7.856-7.216 0-13.972 2.788-19.044 7.856l-16.132 16.136c-5.068 5.064-7.86 11.828-7.86 19.04 0 7.208 2.792 14.2 7.86 19.264L355.9 207.526H26.58C11.732 207.526 0 219.15 0 234.002v22.812c0 14.852 11.732 27.648 26.58 27.648h330.496L252.248 388.926c-5.068 5.072-7.86 11.652-7.86 18.864 0 7.204 2.792 13.88 7.86 18.948l16.132 16.084c5.072 5.072 11.828 7.836 19.044 7.836 7.208 0 13.968-2.8 19.04-7.872l177.68-177.68c5.084-5.088 7.88-11.88 7.86-19.1.016-7.244-2.776-14.04-7.864-19.12z"
            />
          </svg>
          <span>
            Super Secret Admin Page
          </span>
        </div>
      </a>
    </div>
  </div>
</div>
`;
